components/root-actions/MoveAuthoredContributions.js

```diff
@@ -0,0 +1,315 @@
+import MessageBoxGraphqlError from '../MessageBoxGraphqlError';
+import StyledButton from '../StyledButton';
+import StyledCheckbox from '../StyledCheckbox';
+import StyledInputField from '../StyledInputField';
+import StyledLink from '../StyledLink';
+import StyledSelect from '../StyledSelect';
+import StyledTag from '../StyledTag';
+import { Label, P, Span } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const moveOrdersFieldsFragment = gqlV2/* GraphQL */ `
+  fragment MoveOrdersFields on Order {
+    id
+    legacyId
+    description
+    createdAt
+    amount {
+      valueInCents
+      currency
+    }
+    fromAccount {
+      id
+      name
+      slug
+      isIncognito
+      imageUrl(height: 48)
+    }
+    toAccount {
+      id
+      slug
+      name
+    }
+  }
+`;
+
+const ordersQuery = gqlV2/* GraphQL */ `
+  query AuthoredOrdersRootQuery($account: AccountReferenceInput!) {
+    orders(account: $account, filter: OUTGOING, limit: 100, includeIncognito: true) {
+      nodes {
+        ...MoveOrdersFields
+      }
+    }
+  }
+  ${moveOrdersFieldsFragment}
+`;
+
+const moveOrdersMutation = gqlV2/* GraphQL */ `
+  mutation MoveOrdersMutation(
+    $orders: [OrderReferenceInput!]!
+    $fromAccount: AccountReferenceInput!
+    $makeIncognito: Boolean
+  ) {
+    moveOrders(orders: $orders, fromAccount: $fromAccount, makeIncognito: $makeIncognito) {
+      ...MoveOrdersFields
+    }
+  }
+  ${moveOrdersFieldsFragment}
+`;
+
+const getOrdersOptionsFromData = (intl, data) => {
+  if (!data?.orders) {
+    return [];
+  }
+
+  return data.orders.nodes.map(order => {
+    const date = intl.formatDate(order.createdAt);
+    const amount = formatCurrency(order.amount.valueInCents, order.amount.currency, { locale: intl.locale });
+    return {
+      value: order,
+      label: `${date} - ${amount} contribution to @${order.toAccount.slug} (#${order.legacyId})`,
+    };
+  });
+};
+
+const getCallToAction = (selectedOrdersOptions, newFromAccount) => {
+  if (newFromAccount?.isIncognito) {
+    return `Mark ${selectedOrdersOptions.length} contributions as incognito`;
+  } else {
+    const base = `Move ${selectedOrdersOptions.length} contributions`;
+    return !newFromAccount ? base : `${base} to @${newFromAccount.slug}`;
+  }
+};
+
+const getToAccountCustomOptions = fromAccount => {
+  if (!fromAccount) {
+    return [];
+  }
+
+  const incognitoLabel = `@${fromAccount.slug}'s incognito profile`;
+  return [
+    {
+      [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
+      value: fromAccount,
+    },
+    {
+      [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
+      label: incognitoLabel,
+      value: { name: incognitoLabel, useIncognitoProfile: true, isIncognito: true },
+    },
+  ];
+};
+
+const formatOrderOption = (option, intl) => {
+  const order = option.value;
+  return (
+    <Flex alignItems="center" title={order.description}>
+      <Avatar collective={order.fromAccount} size={24} />
```

Comment:
```suggestion
```
Is this `Avatar` needed? Just wondering since we only show the orders from the selected account I don't see any changes to this `Avatar`. So if we have a long list of orders we will see a repeating list of avatars. 🤔

Reply:
It's there because we include user's incognito contributions, it's a handy way to differentiate between regular orders and incognito orders.

Reply:
Ah that makes sense. 👍🏽